packages/xo-web/src/xo-app/hub/recipes/recipe.js

```diff
@@ -51,15 +51,37 @@ export default decorate([
           size: 'medium',
         })
 
-        const { masterName, nbNodes, network, sr, sshKey } = recipeParams
+        const {
+          gatewayIpAddress,
+          masterIpAddress,
+          masterName,
+          nbNodes,
+          network,
+          networkMask,
+          sr,
+          sshKey,
+          staticIpAddress,
+        } = recipeParams
+
+        let workerNodeIpAddresses
+        if (staticIpAddress) {
+          workerNodeIpAddresses = []
+          for (let i = 0; i < nbNodes; i++) {
+            workerNodeIpAddresses[i] = recipeParams[`workerIpAddress${i + 1}`]
+          }
+        }
 
         markRecipeAsCreating(RECIPE_INFO.id)
         const tag = await createKubernetesCluster({
+          gatewayIpAddress,
+          masterIpAddress,
           masterName,
           nbNodes: +nbNodes,
           network: network.id,
+          networkMask,
           sr: sr.id,
           sshKey,
+          workerNodeIpAddresses,
```

Comment:
```suggestion
          workerNodeIpAddresses: Object.values(recipeParams.workerIpAddress),
```

If you take into account the suggestion above, you can use this and remove the assignment of the array.